### PR TITLE
bump chia_rs to 0.7.0

### PR DESCRIPTION
**Pull request by arvidn**
* transition using sized ints and sized bytes from `chia_rs`. This is a step towards removing circular dependencies
* transition `ConsensusConstants` to use the type from `chia_rs`. This is in preparation for porting code that needs these constants to rust

### CODE REVIEW

File: chia/_tests/wallet/wallet_block_tools.py
Comment (altendky) [inline chia/_tests/wallet/wallet_block_tools.py:40]: ?

```suggestion
DEFAULT_VDF_INFO = VDFInfo(bytes32([0] * 32), uint64(1), ClassgroupElement(bytes100(100)))
```

File: chia/types/blockchain_format/proof_of_space.py
Comment (altendky) [inline chia/types/blockchain_format/proof_of_space.py:89]: ?

```suggestion
    prefix_bits: uint8 = constants.NUMBER_ZERO_BITS_PLOT_FILTER
```

**Review by Rigidity — DISMISSED** [review]
Did a few sanity checks, looks good to me. But I wrote a lot of the changes, so my review should be taken with a grain of salt. Need someone else to take a look too.

File: chia/wallet/cat_wallet/cat_wallet.py
Comment (altendky) [inline chia/wallet/cat_wallet/cat_wallet.py:88]: This is losing the type and message of the raised exception since `uint64()` would raise.

File: chia/types/spend_bundle.py
Comment (altendky) [inline chia/types/spend_bundle.py:83]: This is losing the type and message of the raised exception since `uint64()` would raise.

File: chia/wallet/trading/offer.py
Comment (altendky) [inline chia/wallet/trading/offer.py:155]: Similarly, what is the effect of raising here when less than 0 compared to existing behavior?

File: chia/wallet/trading/offer.py
Comment (altendky) [inline chia/wallet/trading/offer.py:173]: Similarly, what is the effect of raising here when less than 0 compared to existing behavior?

File: chia/rpc/wallet_rpc_api.py
Comment (altendky) [inline chia/rpc/wallet_rpc_api.py:1280]: What is up here that we are "losing" the `Coin` type?  I'm guessing it was a lie before?  But then below `.values()` would be an actual behavior change.

**Review by altendky — CHANGES_REQUESTED** [review]

File: chia/types/blockchain_format/proof_of_space.py
Comment (Rigidity) [reply chia/types/blockchain_format/proof_of_space.py:89]: It's possible for this to go negative, which is why there's a `max` at the end.

File: chia/rpc/wallet_rpc_api.py
Comment (Rigidity) [reply chia/rpc/wallet_rpc_api.py:1280]: There was a mistake in our chia_rs type stub previously where `Dict` was not imported, so it was treated as `Any`. It should never have allowed `Coin` here.

`.values()` was also to satisfy Mypy iirc. I'll look more into it, since I don't intend to change behavior.

File: chia/_tests/wallet/wallet_block_tools.py
Comment (Rigidity) [reply chia/_tests/wallet/wallet_block_tools.py:40]: This is necessary, since `bytes100` does not accept `int` as an argument, whereas `bytes` does. I tried your change and it fails CI because of this.

File: chia/wallet/cat_wallet/cat_wallet.py
Comment (Rigidity) [reply chia/wallet/cat_wallet/cat_wallet.py:88]: Yes, good point. I'll update to use `int` until the end.

**Review by altendky — DISMISSED** [review]
Should `chia.util.byte_types` be removed?  Or at least `SizedBytes` from it?

**Comment by Rigidity** [comment]
> Should `chia.util.byte_types` be removed? Or at least `SizedBytes` from it?

Yep, good catch

**Comment by Rigidity** [comment]
`test_struct_stream` is also not necessary anymore, since it was moved to `chia_rs`
https://github.com/Chia-Network/chia_rs/blob/main/tests/test_struct_stream.py

File: chia/_tests/wallet/wallet_block_tools.py
Comment (altendky) [reply chia/_tests/wallet/wallet_block_tools.py:40]: Mm, I did leave that out of the hinting.  Would have been just `SupportsIndex` perhaps, without the `Iterable[]` bit.

https://github.com/Chia-Network/chia-blockchain/blob/1821d6b575d3b588d56495d8a4a0ed445d7b4aed/chia/util/byte_types.py#L30

If we want to continue not allowing that behavior then maybe the `[0] * 100` pattern would be better.  Or `0 for _ in range(100)`.  Or really, `bytes100.zeroes()` and `bytes100.pad()`.  But I didn't get around to adding that before you got around to deleting the whole class. `:]`

File: chia/types/blockchain_format/proof_of_space.py
Comment (altendky) [reply chia/types/blockchain_format/proof_of_space.py:89]: Are we still working towards the fixed width types not subclassing `int`?  Though I guess if we are then a complaint would be made and addressed when we did that.

Comment (Rigidity) [reply chia/types/blockchain_format/proof_of_space.py:89]: I didn't know that was a goal, but that's out of scope for this PR either way

File: chia/_tests/wallet/wallet_block_tools.py
Comment (Rigidity) [reply chia/_tests/wallet/wallet_block_tools.py:40]: I didn't delete the class, it was just moved to `chia_rs`. I think this is low priority, but feel free to add it to the type hint there if you want.

File: chia/types/blockchain_format/proof_of_space.py
Comment (arvidn) [reply chia/types/blockchain_format/proof_of_space.py:89]: One approach I was exploring in porting the fixed size int types to rust required first making them not inherit from `int`, because pyo3 doesn't support that. The new strategy is simpler. Any code written in rust will already use fixed size ints and I think it's fine if python code uses `int`

File: chia/_tests/wallet/wallet_block_tools.py
Comment (arvidn) [reply chia/_tests/wallet/wallet_block_tools.py:40]: I think `[0] * 100` is a bit clearer.

File: chia/rpc/wallet_rpc_api.py
Comment (arvidn) [reply chia/rpc/wallet_rpc_api.py:1280]: it looks suspicious to me. surely the "excluded_coins" *could* be a list of coins, JSON supports that, right? It seems that regardless of whether the typing is correct or not, calling `.values()` below is a change in behaviour.

Comment (arvidn) [reply chia/rpc/wallet_rpc_api.py:1280]: as far as I can tell, a JSON dict is typed as `Any` in the `chia_rs` type stub. iirc, some older versions of mypy doesn't support recursive definitions, like `JSON = Union[List[JSON], Dict[str, JSON], int, str]`

File: chia/types/blockchain_format/proof_of_space.py
Comment (arvidn) [inline chia/types/blockchain_format/proof_of_space.py:88]: this looks like a gratuitous change. I don't see any call sites adjusted in this patch because of this

File: chia/rpc/wallet_rpc_api.py
Comment (arvidn) [reply chia/rpc/wallet_rpc_api.py:1280]: as I suspected. We don't have test coverage for the `.values()` line. I will revert this change

**Review by Rigidity — DISMISSED** [review]

File: chia/types/blockchain_format/proof_of_space.py
Comment (Rigidity) [reply chia/types/blockchain_format/proof_of_space.py]: Good point

File: chia/rpc/wallet_rpc_api.py
Comment (Rigidity) [reply chia/rpc/wallet_rpc_api.py]: My bad, good catch

File: chia/types/blockchain_format/proof_of_space.py
Comment (Rigidity) [reply chia/types/blockchain_format/proof_of_space.py]: Figured it's ideal to have stricter typing for this, but I guess there's a million other places where this is the case anyways.

File: chia/types/blockchain_format/proof_of_space.py
Comment (altendky) [reply chia/types/blockchain_format/proof_of_space.py]: Do we need to convert it?  Remember that `uint8(5) + uint8(6)` gives you an `int` since we didn't set up any 'type retention' (i just made that term up, afaik) on these.

File: chia/wallet/cat_wallet/cat_wallet.py
Comment (altendky) [reply chia/wallet/cat_wallet/cat_wallet.py]: As above, is any conversion needed?

File: chia/types/blockchain_format/proof_of_space.py
Comment (arvidn) [reply chia/types/blockchain_format/proof_of_space.py]: right, I keep forgetting this. I think it's quite subtle, and causes `a += 1` to fail mypy (iirc) because `a` changes type.
I don't have a strong opinion, but I think it's quite reasonable to show that we want the `int` explicitly

File: chia/wallet/cat_wallet/cat_wallet.py
Comment (arvidn) [reply chia/wallet/cat_wallet/cat_wallet.py]: you're right that the operators return `int`, so it's not necessary. Do you think I should revert these?

File: chia/types/blockchain_format/proof_of_space.py
Comment (altendky) [reply chia/types/blockchain_format/proof_of_space.py]: image

Quite so.  Alrighty, we can move forward.  Thanks.

**Review by Rigidity — DISMISSED** [review]
lgtm

**Review by altendky — APPROVED** [review]

**Comment by Rigidity** [comment]
I made the change you requested, just in a separate commit. If I had pressed a button to do it it would have also dismissed the review. Same end result.

**Comment by altendky** [comment]
In so much as we don't have written policies for this but it has been discussed...  Review dismissal is reserved for when someone is non-responsive and they are known to be unreachable.  Not for 'i feel i did what they said'.  Generally the manager of the person whose review is being dismissed would do the dismissal.  I was still chatting with Arvid about some stuff.  You can wait another hour for your PR to get approved.  Additionally, do not resolve other people's conversations.  Everyone that has pointed out a concern in the conversation should be in agreement prior to resolution.

**Comment by altendky** [comment]
> I made the change you requested, just in a separate commit. If I had pressed a button to do it it would have also dismissed the review. Same end result.

This is also incorrect.  If I had made a suggestion using the GitHub feature and you had clicked the commit button...  Yes, it would have resolved the conversation which is ...  not the best in my mind but also not terrible since you committed exactly code that I had looked at, written, and suggested.  No, it would not have dismissed my review.  So, this is both a different case (I did not provide committable suggestions in all cases) and the end result is also not the same.

**Comment by Rigidity** [comment]
> > I made the change you requested, just in a separate commit. If I had pressed a button to do it it would have also dismissed the review. Same end result.
> 
> This is also incorrect. If I had made a suggestion using the GitHub feature and you had clicked the commit button... Yes, it would have resolved the conversation which is ... not the best in my mind but also not terrible since you committed exactly code that I had looked at, written, and suggested. No, it would not have dismissed my review. So, this is both a different case (I did not provide committable suggestions in all cases) and the end result is also not the same.

I'm not going to sit here and argue this completely inconsequential thing with you. All of the feedback has been implemented in the PR. If you still have _OTHER_ changes requested, please review the PR accordingly. Tone your attitude back. Thanks.

**Review by Rigidity — APPROVED** [review]

**Comment by Rigidity** [comment]
Re the coverage diff, I don't believe we've changed that other than for the replace function, moving it outside of the `ConsensusConstants` class. So presumably no new code is left untested?

**Comment by arvidn** [comment]
> Re the coverage diff, I don't believe we've changed that other than for the replace function, moving it outside of the ConsensusConstants class. So presumably no new code is left untested?

That's right. I think it's fine to ignore the coverage diff. I will make a separate PR to add test coverage for this function

**Review by emlowe — APPROVED** [review]

**Pull request merged**
